src/dnscap.c

```diff
@@ -1132,8 +1132,8 @@ parse_args(int argc, char *argv[]) {
 		usage("the -b option is incompatible with -d and -g");
         if (sample && chooseSidesResponse)
         {
-                if (!((dir_wanted & DIR_INITIATE) != 0) && ((dir_wanted & DIR_RESPONSE) != 0))
-                        usage("the -q option is incompatible with -s r");
+		if(((dir_wanted & DIR_RESPONSE) != 0) && ((dir_wanted & DIR_INITIATE) == 0))
+			usage("the -q option is incompatible with -s r");
```

Comment:
On second glance I do not see the reason to have `chooseSidesResponse`, `dir_wanted` is set to both direction at start and you really only need to check if `DIR_RESPONSE` is not set.